owncloud/core
# Add force option to delete user home folder

Add a force option to the user delete command
in order to delete the users home folder if
exists.

Signed-off-by: Sujith H <sharidasan@owncloud.com>



## Description

Add force option for the `user:delete` command, to delete the home folder of the user if available.

## Related Issue




https://github.com/owncloud/enterprise/issues/2969

## Motivation and Context
Its been noticed that when users upgrade from 9.1.5 to 10 of core, the accounts table does not have the user details and neither the home entry added to it. Now if the backend had deleted a user before upgrade then oC admin will not be able to delete the oc user mapped to the ldap backend. Because the entry is missing in the accounts table. Under such condition it would be handy to delete the home folder of those users (i.e, for missing backend users).

## How Has This Been Tested?



- Install oC 9.1.5
- Create LDAP users
- Now enable user_ldap app
- Configure it with the oC as admin
- Make sure the LDAP users are synced. You can see the result in the Users page.
- Now try to login to few users who are mapped with the LDAP.
- Delete the users in the LDAP.
- Upgrade to oC 10.4.0
- Now try to run `user:delete` with uid and --force, where uid is the user unavailable in the LDAP or say the deleted one.
- The home folder ( i.e, the remains ) should be deleted.

## Screenshots (if appropriate):

## Types of changes


## Checklist:
